Add content/en/role-based access control(RBAC) (#1503)
cncf/glossary: content/en/role-based-access-control.md

---
title: Role-Based Access Control (RBAC)
status: Feedback Appreciated
category: concept
---

## What it is

> The method of regulating systems and resources access based on individual users' roles within an organization.

Reviewer: ```suggestion
Role-Based Access Control (RBAC) is a method to regulate system and resource access based on individual users' roles within an organization.
```

> 

Reviewer: ```suggestion
Each job function within an organization has a different role with specific permissions. 
For instance, a marketing team member may have permission to view the deals sales has in the pipeline (*view* permission) but won't be allowed to edit them (*write* permission). 
Sales and marketing have different permissions based on their roles, hence role-based access control. 

```

Reviewer: I think an example would help here.
View/write permission might to be the right term. Can you help me out @iamNoah1?

Reviewer 2: @CathPag I like your edit and I think it's more read/write than view/write.

> The method of regulating access to computer or network resources based on individual users' roles within an organization.

Reviewer 3: Hi @juliafmorgado 

I am not sure the current resource targets (`computer or network resources`) of access control fully cover targets that can be handled by RBAC. 

For instance, RBAC can be applied to various resources (objects) in Kubernetes such as Pods or Services.

Author: What about "regulating systems and resources access"?

Reviewer 3: @juliafmorgado 
Yes, it looks fine for me.

## Problem it addresses

> To maintain the security of an organization, it becomes increasingly important to limit unnecessary permissions based on employee roles.

Reviewer: This is not describing the problem. It's a statement of why RBAC is necessary. This section should focus on what would a world without RBAC look like? Ideally, we shouldn't even mention the solution, just focus on the problem.

Author: Thanks for the feedback Cath, I'll update it now.

---
title: Role-Based Access Control (RBAC)
status: Feedback Appreciated
category: concept
---

## What it is

Role-Based Access Control (RBAC) is a method to regulate system and resource access based on individual users' roles within an organization.
Each job function within an organization has a different role with specific permissions. 
For instance, a marketing team member may have permission to view the deals sales has in the pipeline (*read* permission) but won't be allowed to edit them (*write* permission). 
Sales and marketing have different permissions based on their roles, hence role-based access control. 

## Problem it addresses

> Without RBAC, it is easy to lose track of who has access to what. Malicious actors are always in search of vulnerabilities to exploit and a mismanaged account with too many privileges can put an entire system at risk. 

Reviewer 2: true, preventing malicious actors is not the main focus of RBAC. I cannot exactly say if before RBAC the solution was to have separate systems. I would more likely assume that the access levels have been defined on a User level rather than using roles.

> Without RBAC, it is easy to lose track of who has access to what. Malicious actors are always in search of vulnerabilities to exploit and a mismanaged account with too many privileges can put an entire system at risk. 

Reviewer 2: I guess we need some third opinion here. @seokho-son @jihoon-seo @mfosterrox what do you think?

## How it helps

> It simplifies access management and enables more granular control of user actions in software applications.

Reviewer: ```suggestion
RBAC enables granular control over user permissions within software systems. 
Depending on their roles, team members might be able only to view a section, change it, or have no access at all. 
```

> The allocation of features to the right people prevents overprivileged accounts and unauthorized access.

Reviewer: ```suggestion
This granular permission allocation allows various team members with different roles to work on the same system without risking...
```
What does "overprivileged accounts and unauthorized access" mean concretely?

Reviewer 2: I would say that both statements are somehow correct. Overpriviliged would mean that users or groups have more permissions than they need, while unauthorized ones would have fewer permissions than they need?!

Author: As Noah said, overprivileged means that users or groups have more permissions than they need. The same applies to unauthorized access; users or groups have access to systems they shouldn't have.

---
title: Role-Based Access Control (RBAC)
status: Feedback Appreciated
category: concept
---

## What it is

Role-Based Access Control (RBAC) is a method to regulate system and resource access based on individual users' roles within an organization.
Each job function within an organization has a different role with specific permissions. 
For instance, a marketing team member may have permission to view the deals sales has in the pipeline (*read* permission) but won't be allowed to edit them (*write* permission). 
Sales and marketing have different permissions based on their roles, hence role-based access control. 

## Problem it addresses

> Without RBAC, it is easy to lose track of who has access to what. Malicious actors are always in search of vulnerabilities to exploit and a mismanaged account with too many privileges can put an entire system at risk. 

Reviewer 3: I agree with @CathPag and @iamNoah1 
I think we'd better focus on `role` and not `access control` in the problem it address.

following is my suggestion. (please improve it if necessary)

```suggestion
Individually managing access permissions to many users in an organization of various system resources and data is very complex considering that each user will have different set of resources the user can access. For instance, if there are 100,000 users and 1,000 resources in an organization, an administrator needs to check whether each user has authority and responsibility for each resource 100,000 times.
Therefore, it is necessary to simplify the access control according to role groups based on authority and responsibility.
```

---
title: Role-Based Access Control (RBAC)
status: Feedback Appreciated
category: concept
---

## What it is

Role-Based Access Control (RBAC) is a method to regulate system and resource access based on individual users' roles within an organization.
Each job function within an organization has a different role with specific permissions. 
For instance, a marketing team member may have permission to view the deals sales has in the pipeline (*read* permission) but won't be allowed to edit them (*write* permission). 
Sales and marketing have different permissions based on their roles, hence role-based access control. 

## Problem it addresses

> Individually managing access permissions to many users in an organization of various system resources and data is very complex considering that each user will have different set of resources the user can access. For instance, if there are 100,000 users and 1,000 resources in an organization, an administrator needs to check whether each user has authority and responsibility for each resource 100,000 times.

Reviewer: ```suggestion
Individually managing access permissions of multiple users across various system resources and data can be complex.
After all, each user likely needs access to a different set of resources. 
Let's say an organization has 500 developers (users) and 300 microservices with varying permission levels (read, write, and no access). The administrator must ensure each user has the correct permissions for all 500 resources.  
```
I changed the example to a smaller group. 100,000 is a huge organization. Most people work at much smaller companies. 500 developers is still big but might feel more real. We can also decrease that number to 50 (?)

> Therefore, it is necessary to simplify the access control according to role groups based on authority and responsibility.

Reviewer: ```suggestion
RBAC simplifies the process by providing predefined access control based on role groups.
```

## How it helps

> RBAC enables granular control over user permissions within software systems. 

Reviewer: ```suggestion
RBAC provides granular control over user permissions within software systems. 
```

> Depending on their roles, team members might be able only to view a section, change it, or have no access at all. 

Reviewer: ```suggestion
Depending on their roles, team members can modify a section, only view it, or have no access at all. 
```

> This granular permission allocation allows various team members with different roles to work on the same system without risking.

Reviewer: ```suggestion
This granular permission allocation allows team members with different roles to work on the same system while minimizing risk.
```
